content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+    camera: Camera {
+        // renders after / on top of the main camera
+        priority: 1,
+        ..default()
+    },
+    ..default()
+});
+```
+
+"Priority" determines the order cameras are drawn in. The "portal" effect in the [render to textures example above](#render-to-textures) uses priority to render the "portal" camera first, ensuring it is ready to be used by the main camera.
+
+Here is a simple example of two cameras rendering to the same window:
+
+![camera layering](camera_layering.png)
+
+This can be used for things like "custom UI passes", "minimaps", etc.
+
+### Ergonomic Target Size Access
+
+Working with cameras is now also much easier. They now store their [`RenderTarget`] size locally, which makes retrieving the size much simpler:
+
+```rust
+/// Much nicer than needing to look up the size on the target Window or Image manually,
+/// like you had to in previous Bevy versions!
+let target_size = camera.logical_target_size();
+let viewport_size = camera.logical_viewport_size();
+```
+
+This also means that converting world to screen coordinates for a camera is much easier than it used to be:
+
+```rust
+// Bevy 0.7 (old)
+camera.world_to_screen(windows, images, camera_transform, world_position);
+
+// Bevy 0.8 (new)
+camera.world_to_viewport(camera_transform, world_position);
+```
+
+### New Camera Bundles
+
+The old `OrthographicCameraBundle` and `PerspectiveCameraBundle` have been replaced with [`Camera3dBundle`] and [`Camera2dBundle`]. In most cases migration should be as simple as replacing the old names with the new ones.
+
+### No More CameraUiBundle!
+
+Bevy UI now no longer needs a separate camera entity to work. UI "just works" for all camera types and can be enabled or disabled per-camera using the [`UiCameraConfig`] component.
+
+### Custom Render Graphs
+
+The default 2D and 3D [`RenderGraphs`][`RenderGraph`] for each [`Camera`] can be overridden by setting the [`CameraRenderGraph`] component:
+
+```rust
+commands.spawn_bundle(Camera3dBundle {
+    camera_render_graph: CameraRenderGraph::new(some_custom_graph),
+    ..default()
+})
+```
+
+This enables you to draw the camera with whatever custom render logic you need! Note that generally this won't be required: most custom rendering scenarios will be covered by high level [Materials](#new-material-system) or extending the built in render graphs.
+
+### Enabling / Disabling Cameras
+
+If a camera is "active", it will render to its [`RenderTarget`]. To activate or deactivate a camera, set the new `is_active` field:
+
+```rust
+camera.is_active = true;
+```
+
+### RenderLayers
+
+Bevy's existing [`RenderLayers`] system can be used to tell a [`Camera`] to only render entities on specific layers. **Camera Driven Rendering** pairs nicely with this feature. This enables rendering one set of entities to one camera and another set of entities to another camera. We've ported the [`RenderLayers`] system to all entities with [`Visibility`], so this will all Just Work™.
```

Comment:
```suggestion
Bevy's existing [`RenderLayers`] system can be used to tell a [`Camera`] to only render entities on specific layers. **Camera-Driven Rendering** pairs nicely with this feature. This enables rendering one set of entities to one camera and another set of entities to another camera. We've ported the [`RenderLayers`] system to all entities with [`Visibility`], so this will all Just Work™.
```